### PR TITLE
separate early error tests into their own test suite

### PR DESCRIPTION
**Pull request by michaelficarra**
@ikarienator Please either disable or fix the broken tests by adding commits to this branch. I will no longer force-push.

### CODE REVIEW

File: test/destructuring/binding-pattern/array-binding.js [hunk 1]
```diff
@@ -55,11 +55,13 @@ suite("Parser", function () {
 
       testParseFailure("var [a.b]", "Unexpected token [");
       testParseFailure("var ([x])", "Unexpected token (");
+      // TODO: these appear to be valid
```
Comment (michaelficarra) [inline test/destructuring/binding-pattern/array-binding.js]: @ikarienator Can you confirm this?

Comment (ikarienator) [inline test/destructuring/binding-pattern/array-binding.js]: I confirm but it feels like a bug in the spec.

Comment (michaelficarra) [inline test/destructuring/binding-pattern/array-binding.js]: https://bugs.ecmascript.org/show_bug.cgi?id=4095

File: test/destructuring/binding-pattern/object-binding.js [hunk 2]
```diff
@@ -35,6 +35,7 @@ suite("Parser", function () {
         ]))
       );
 
+      // TODO: these appear to be valid
```
Comment (michaelficarra) [inline test/destructuring/binding-pattern/object-binding.js]: https://bugs.ecmascript.org/show_bug.cgi?id=4095

File: test/early-errors.js [hunk 3]
```diff
@@ -0,0 +1,638 @@
...
+    testParseFailure("({a}) => { let a; }", "");
+    testParseFailure("(a) => { const a = 0; }", "");
+    testParseFailure("([a]) => { const a = 0; }", "");
+    testParseFailure("({a}) => { const a = 0; }", "");
+    // If the [Yield] grammar parameter is present on ArrowParameters, it is a Syntax Error if the lexical token sequence matched by CoverParenthesizedExpressionAndArrowParameterList[?Yield] cannot be parsed with no tokens left over using ArrowFormalParameters[Yield, GeneratorParameter] as the goal symbol.
+    // TODO
+    // If the [Yield] grammar parameter is not present on ArrowParameters, it is a Syntax Error if the lexical token sequence matched by CoverParenthesizedExpressionAndArrowParameterList[?Yield] cannot be parsed with no tokens left over using ArrowFormalParameters as the goal symbol.
+    // TODO
+    // All early errors rules for ArrowFormalParameters and its derived productions also apply to CoveredFormalsList of CoverParenthesizedExpressionAndArrowParameterList[?Yield].
+    // TODO
```
Comment (michaelficarra) [inline test/early-errors.js:465]: @ikarienator Can you write tests for these three?

File: test/modules/export.js [hunk 4]
```diff
@@ -191,6 +191,8 @@ suite("Parser", function () {
     testParseModuleFailure('export default default', 'Unexpected token default');
     testParseModuleFailure('export default function', 'Unexpected end of input');
     testParseModuleFailure('export default let', 'Unexpected token let');
+
+    // TODO: these should definitely pass
```
Comment (michaelficarra) [inline test/modules/export.js]: https://github.com/shapesecurity/shift-parser-js/issues/122

File: test/expressions/yield-expression.js [hunk 5]
```diff
@@ -35,36 +35,34 @@ suite("Parser", function () {
...
-    testParse('function *a(){yield "a"}', yde, new Shift.LiteralStringExpression('a'));
-    testParse('function *a(){yield a}', yde, new Shift.IdentifierExpression(new Shift.Identifier('a')));
-    testParse('function *a(){yield+0}', yde, new Shift.PrefixExpression('+', new Shift.LiteralNumericExpression(0)));
-    testParse('function *a(){yield-0}', yde, new Shift.PrefixExpression('-', new Shift.LiteralNumericExpression(0)));
-    testParse('function *a(){yield 2e308}', yde, new Shift.LiteralInfinityExpression());
-    testParse('function *a(){yield(0)}', yde, new Shift.LiteralNumericExpression(0));
-    testParse('function *a(){yield/a/}', yde, new Shift.LiteralRegExpExpression('a', ''));
-    testParse('function *a(){yield/=3/}', yde, new Shift.LiteralRegExpExpression('=3', ''));
-    testParse('function *a(){yield class{}}', yde, new Shift.ClassExpression(null, null, []));
+    // TODO: figure out which early error applies here
```
Comment (michaelficarra) [inline test/expressions/yield-expression.js]: @ikarienator can you put these failure tests in the appropriate place in `early-errors.js`?

Comment (ikarienator) [inline test/expressions/yield-expression.js]: None of them are early errors.

**Comment by ikarienator** [comment]
I'd say the only early errors left are impossible to handle without validator. Shall we merge this and left them as TODOs?

**Pull request merged**